drivers/soundwire/sysfs_slave_dpn.c

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright(c) 2015-2020 Intel Corporation.
+
+#include <linux/device.h>
+#include <linux/mod_devicetable.h>
+#include <linux/slab.h>
+#include <linux/sysfs.h>
+#include <linux/soundwire/sdw.h>
+#include <linux/soundwire/sdw_type.h>
+#include "bus.h"
+#include "sysfs_local.h"
+
+/*
+ * DP-N properties
+ */
+
+#define to_dpn_prop(a, b) (a).b
+
+#define sdw_dpn_attr(N, dir, field, format_string)			\
+static ssize_t								\
+dp##N##_##dir##_##field##_show(struct device *dev,			\
+			       struct device_attribute *attr,		\
+			       char *buf)				\
+{									\
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);			\
+	struct sdw_dpn_prop *dpn =					\
+			to_dpn_prop(slave->prop, dir##_dpn_prop);	\
+	unsigned long mask;						\
+	int bit;							\
+	int i;								\
+									\
+	/* convert src to source */					\
+	if (dpn == slave->prop.src_dpn_prop)				\
+		mask = slave->prop.source_ports;			\
+	else								\
+		mask = slave->prop.sink_ports;				\
+	i = 0;								\
+	for_each_set_bit(bit, &mask, 32) {				\
+		if (bit == N) {						\
+			return sprintf(buf, format_string,		\
+				       dpn[i].field);			\
+		}							\
+		i++;							\
+	}								\
+	return -EINVAL;							\
+}									\
+static DEVICE_ATTR_RO(dp##N##_##dir##_##field);				\
```

Comment:
I think previous patches didn't have a semicolon in the above line, which then allowed them to add semicolons to lines below. I'd do that here too.

Reply:
no longer used